front_matter/json.ts

```diff
@@ -32,7 +32,9 @@ export type { Extract };
  * @param text The text to extract JSON front matter from.
  * @returns The extracted JSON front matter and body content.
  */
-export function extract<T>(text: string): Extract<T> {
+export function extract<T extends Record<string, unknown>>(
+  text: string,
+): Extract<T> {
```

Comment:
Similarly to any.ts, can we keep the existing type? `T` can be non Record type, and this change breaks such case:

```ts
import { extract } from "@std/front-matter/json";

console.log(extract<boolean>(`---json
true
---

aaa
`));
```